pkgs/node/src/chain.zig

```diff
@@ -47,10 +47,12 @@ pub const ProducedBlock = struct {
 
 pub const BeamChain = struct {
     config: configs.ChainConfig,
+    anchor_state: *types.BeamState,
+
     forkChoice: fcFactory.ForkChoice,
     allocator: Allocator,
     // from finalized onwards to recent
-    states: std.AutoHashMap(types.Root, types.BeamState),
+    states: std.AutoHashMap(types.Root, *types.BeamState),
```

Comment:
can we also make it a *const types.BeamState map?

Reply:
This should not be possible as the map is the owner here, if I make it `*const types.BeamState` I won't be able to free the memory as `deinit()` cannot be called on `const`